libs/classpath_resolver/src/resolver.rs

```diff
@@ -0,0 +1,126 @@
+use std::{
+    fs,
+    io::{self, Write},
+    path::{Path, PathBuf},
+    process::Command,
+    result,
+};
+
+#[derive(Debug)]
+pub enum WorkspaceType {
+    Maven,
+    Gradle,
+}
+
+pub struct ClasspathResolver {
+    workspace_type: Option<WorkspaceType>,
+    workspace_root: PathBuf,
+    classpath: Vec<PathBuf>,
+}
+
+impl ClasspathResolver {
+    pub fn new_with_type(
+        workspace_root: PathBuf,
+        workspace_type: Option<WorkspaceType>,
+    ) -> ClasspathResolver {
+        let classpath = match workspace_type {
+            Some(WorkspaceType::Maven) => resolve_maven_classpath(&workspace_root),
+            Some(WorkspaceType::Gradle) => resolve_gradle_classpath(&workspace_root),
+            None => Ok(vec![]),
+        };
+
+        ClasspathResolver {
+            workspace_type,
+            workspace_root,
+            classpath: classpath.unwrap_or(vec![]),
+        }
+    }
+
+    pub fn new(workspace_root: PathBuf) -> ClasspathResolver {
+        let workspace_type = detect_workspace_type(&workspace_root);
+
+        ClasspathResolver::new_with_type(workspace_root, workspace_type)
+    }
+}
+
+fn detect_workspace_type(workspace_root: &PathBuf) -> Option<WorkspaceType> {
+    if workspace_root.join("pom.xml").exists() {
+        return Some(WorkspaceType::Maven);
+    } else if workspace_root.join("build.gradle").exists()
+        || workspace_root.join("build.gradle.kts").exists()
```

Comment:
should we use "settings.gradle" instead since settings leave in the root always but not build e.g gradle mono repo setup